systemservices/marketplace/src/tasks/publishServiceVersion.ts

```diff
@@ -1,16 +1,21 @@
 import { TaskInputs, TaskOutputs } from "mesg-js/lib/service"
 import { Marketplace } from "../contracts/Marketplace"
 import { asciiToHex, CreateTransaction, hashToHex } from "../contracts/utils";
+import { Manifest } from "../types/manifest";
 
 export default (
   marketplace: Marketplace,
   createTransaction: CreateTransaction
 ) => async (inputs: TaskInputs, outputs: TaskOutputs): Promise<void> => {
   try {
+    const manifestHash = await uploadManifest({
+      version: '1',
+      service: inputs.service
+    })
     const transactionData = marketplace.methods.publishServiceVersion(
-      asciiToHex(inputs.sid),
-      asciiToHex(inputs.manifest),
-      asciiToHex(inputs.manifestProtocol)
+      asciiToHex(inputs.service.definition.sid),
```

Comment:
This is super dangerous...
Service is of type Object without any more detailed in the mesg.yml.
I will recommend to at least specify the sid in the mesg.yml and validate it with the json schema of the other PR.

Reply:
i add the sid in the mesg.yml in commit https://github.com/mesg-foundation/core/pull/837/commits/e2d724eeab8ab298fbcd1b088c6f740b3ddd1bd9